indra/newview/llmeshrepository.cpp

```diff
@@ -4241,16 +4301,17 @@ S32 LLMeshRepository::loadMesh(LLVOVolume* vobj, const LLVolumeParams& mesh_para
         mesh_load_map::iterator iter = mLoadingMeshes[new_lod].find(mesh_id);
         if (iter != mLoadingMeshes[new_lod].end())
         { //request pending for this mesh, append volume id to list
-            auto it = std::find(iter->second.begin(), iter->second.end(), vobj);
-            if (it == iter->second.end()) {
-                iter->second.push_back(vobj);
+            auto it = std::find(iter->second.mVolumes.begin(), iter->second.mVolumes.end(), vobj);
+            if (it == iter->second.mVolumes.end()) {
+                iter->second.addVolume(vobj);
             }
         }
         else
         {
             //first request for this mesh
-            mLoadingMeshes[new_lod][mesh_id].push_back(vobj);
-            mPendingRequests.emplace_back(new PendingRequestLOD(mesh_params, new_lod));
+            std::shared_ptr<PendingRequestBase> request(new PendingRequestLOD(mesh_params, new_lod));
+            mPendingRequests.emplace_back(request);
+            mLoadingMeshes[new_lod][mesh_id].initData(vobj, request);
```

Comment:
Very ugly, even unreliable if something start making copies, but I didn't want to remake it entirely so ended up passing weak ptr over and using it to set dependency.